reference/commands/consumer/install.rst

```diff
@@ -155,27 +155,34 @@ generators.
 
 1. ``config_options()``
 2. ``configure()``
-3. ``requirements()``
-4. ``package_id()``
-5. ``package_info()``
-6. ``deploy()``
+3. ``layout()``
+4. ``requirements()``
+5. ``package_id()``
+6. ``validate()``
+7. ``validate_build()``
```

Comment:
@memsharded are you sure? Why validate_build() would be called when we don't build?

Reply:
`--build missing` ?

Reply:
this list doesn't refer to --build missing, it's the list below.

Reply:
Completely sure, both methods are called sequentially: 
```python
      if hasattr(conanfile, "validate") and callable(conanfile.validate):
            with conanfile_exception_formatter(str(conanfile), "validate"):
                try:
                    conanfile.validate()
                    # FIXME: this shouldn't be necessary in Conan 2.0
                    conanfile._conan_dependencies = None
                except ConanInvalidConfiguration as e:
                    conanfile.info.invalid = str(e)

        if hasattr(conanfile, "validate_build") and callable(conanfile.validate_build):
            with conanfile_exception_formatter(str(conanfile), "validate_build"):
                try:
                    conanfile.validate_build()
                except ConanInvalidConfiguration as e:
                    # This 'cant_build' will be ignored if we don't have to build the node.
                    node.cant_build = str(e)
```

Reply:
ok, it's just weird to have a `validate_build()` method name which is called even when we don't build.

Reply:
It is perfectly reasonable to know if a package can be built or not with the current configuration, even if you are not building it, for example in ``conan info`` (that never builts from source). It is very valuable information, very cheap to compute, it makes sense to have the information available.